### PR TITLE
Persist invoking idle through disconnects

### PR DESCRIPTION
**Pull request by loudnate**
Let me know if I'm misunderstanding the concept of the idle trigger here.

### CODE REVIEW

File: RileyLink/RileyLinkBLEDevice.m [hunk 1]
```diff
@@ -177,7 +177,22 @@ - (RileyLinkState) state {
   return rval;
 }
 
-- (void) didDisconnect:(NSError*)error {
+- (void)connectionStateDidChange:(NSError *)error
+{
+  switch (self.peripheral.state) {
```
Comment (loudnate) [inline RileyLink/RileyLinkBLEDevice.m:182]: Yep, I changed my global settings to 2 spaces just for RLIOS dev.

Comment (ps2) [inline RileyLink/RileyLinkBLEDevice.m:182]: :)

**Comment by ps2** [comment]
I _think_ this is ok. Yes we could lose an idle response while disconnected.  Riley goes in and out of range all the time, and I haven't noticed I'm missing data because of it, but we're on dex right now, so it wouldn't be as visible.  The only danger to this is getting back an extra response from an interrupted idle when you reconnect and issue a rx on top of the rx that was already running.

**Comment by loudnate** [comment]
That's a good point. Should we guard `onIdle` with `&& runningIdle == NO` then?

``` objc
- (void) onIdle {
   if (idleListeningEnabled) {
     runningIdle = YES;
     NSLog(@"Starting idle RX");
     GetPacketCmd *cmd = [[GetPacketCmd alloc] init];
     cmd.listenChannel = idleListenChannel;
     cmd.timeoutMS = 60 * 1000;
     [self issueCommand:cmd];
   }
 }
```

**Comment by loudnate** [comment]
We might also want to dispatch to the main queue here before accessing `idleListeningEnabled`, since that's where all the BLE delegate calls are running.

``` objc
- (void) enableIdleListeningOnChannel:(uint8_t)channel
```

**Comment by ps2** [comment]
No, that's pretty much already guarded for elsewhere (we could guard inside there as well or alternately).  
You mark runningIdle = NO when we disconnect, but it's possible that the rx command is still running on the RL when we reconnect, and issue another RX.

**Comment by loudnate** [comment]
Will a second `GetPacketCmd` interrupt the first?

**Comment by ps2** [comment]
Yes, and will generate a SubgRfspyErrorCmdInterrupted response.

**Comment by loudnate** [comment]
Ah, so that cycle of interruption would continue then. What about sending a reset command on reconnect, if `runningIdle == YES`?

**Comment by ps2** [comment]
Then we lose tuning info.  If we do observe it getting into a loop (I'm not sure it will), then we can look at catching the CmdInterrupted earlier, and not leaving idle state.

**Comment by loudnate** [comment]
I can try and force the contrived case and see how it recovers.

**Pull request merged**